app/Commands/Command.php

```diff
@@ -13,7 +16,12 @@ abstract class Command extends BaseCommand
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $this->presenter = resolve(PresenterContract::class, [$input, $output]); // @phpstan-ignore-line
+        try {
+            $this->presenter = resolve(PresenterContract::class); // @phpstan-ignore-line
+        } catch (BindingResolutionException) {
+            $this->presenter = new Presenter($input, $output);
+            $this->app->singleton(PresenterContract::class, fn (Application $app) => $this->presenter);
+        }
```

Comment:
ℹ️ Ensures that the same instance of the `Presenter` service is always retrieved (otherwise we'd lose the current shell)